app/assets/javascripts/admin/order_cycles/controllers/order_cycle_exchanges_controller.js.coffee

```diff
@@ -29,13 +28,29 @@ angular.module('admin.orderCycles')
       OrderCycle.removeExchangeFee(exchange, index)
       $scope.order_cycle_form.$dirty = true
 
-    $scope.addSupplier = ($event) ->
-      $event.preventDefault()
-      OrderCycle.addSupplier($scope.new_supplier_id)
-
-    $scope.addDistributor = ($event) ->
-      $event.preventDefault()
-      OrderCycle.addDistributor($scope.new_distributor_id)
-
     $scope.removeDistributionOfVariant = (variant_id) ->
       OrderCycle.removeDistributionOfVariant(variant_id)
+
+    $scope.loadExchangeProducts = (scope, exchange) ->
+      return if scope.enterprises[exchange.enterprise_id].supplied_products_fetched?
+      scope.enterprises[exchange.enterprise_id].supplied_products_fetched = true
```

Comment:
This passing of scope objects between two controllers looks pretty funky. I think if the controllers are on the same page and nested in the same elements they can probably already share a scope? 

Otherwise generally if you need to pass things between multiple controllers, it's better to create a service that holds the shared data and have each controller interact with it.

Reply:
yes, thanks for flagging this.
I was having some trouble with some scopes not having the info I needed. but now it all works well just using $scope. New commit: 217ae774864f3a0d51df5be48744af3010fa6082